operator/pkg/awsprovider/iam/reconciler.go

```diff
@@ -0,0 +1,211 @@
+	}
+	for _, policy := range kitNodeRolePolicies {
+		if err := role.attachPolicy(ctx, policy, KitNodeRoleNameFor(controlPlane.ClusterName())); err != nil {
+			return fmt.Errorf("attaching policies to role, %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (c *Controller) Finalize(ctx context.Context, controlPlane *apis.ControlPlane) error {
+	_, err := c.iam.RemoveRoleFromInstanceProfileWithContext(ctx, &iam.RemoveRoleFromInstanceProfileInput{
+		InstanceProfileName: aws.String(KitNodeInstanceProfileNameFor(controlPlane.ClusterName())),
+		RoleName:            aws.String(KitNodeRoleNameFor(controlPlane.ClusterName())),
+	})
+	if err != nil && !errors.IsIAMObjectDoNotExist(err) {
+		return fmt.Errorf("removing role from instance profile, %w", err)
+	}
+	_, err = c.iam.DeleteInstanceProfileWithContext(ctx, &iam.DeleteInstanceProfileInput{
+		InstanceProfileName: aws.String(KitNodeInstanceProfileNameFor(controlPlane.ClusterName())),
+	})
+	if err != nil && !errors.IsIAMObjectDoNotExist(err) {
+		return fmt.Errorf("deleting instance profile, %w", err)
+	}
+
+	for _, policy := range kitNodeRolePolicies {
+		if _, err = c.iam.DetachRolePolicyWithContext(ctx, &iam.DetachRolePolicyInput{
+			PolicyArn: aws.String(policy),
+			RoleName:  aws.String(KitNodeRoleNameFor(controlPlane.ClusterName())),
+		}); err != nil {
+			return fmt.Errorf("detaching policy from role, %w", err)
+		}
+	}
+	_, err = c.iam.DeleteRoleWithContext(ctx, &iam.DeleteRoleInput{
+		RoleName: aws.String(KitNodeRoleNameFor(controlPlane.ClusterName())),
+	})
+	if err != nil && !errors.IsIAMObjectDoNotExist(err) {
+		return fmt.Errorf("deleting role, %w", err)
+	}
+	zap.S().Infof("[%s] Deleted IAM Role %v and instance profile",
+		controlPlane.ClusterName(), KitNodeRoleNameFor(controlPlane.ClusterName()))
+	return nil
+}
+
+func (c *Controller) getRole(ctx context.Context, clusterName string) (*role, error) {
+	roleOutput, err := c.iam.GetRoleWithContext(ctx, &iam.GetRoleInput{
+		RoleName: aws.String(KitNodeRoleNameFor(clusterName)),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("getting iam role %v, %w", KitNodeRoleNameFor(clusterName), err)
+	}
+	return &role{iam: c.iam, Role: roleOutput.Role}, nil
+}
+
+func (c *Controller) createRole(ctx context.Context, clusterName string) (*role, error) {
+	roleOutput, err := c.iam.CreateRole(&iam.CreateRoleInput{
+		AssumeRolePolicyDocument: aws.String(assumeRolePolicyDocument),
+		Description:              aws.String("Role assumed by dataplane nodes created by KIT operated"),
+		RoleName:                 aws.String(KitNodeRoleNameFor(clusterName)),
+		Tags:                     generateRoleTags(clusterName),
+	})
+	return &role{iam: c.iam, Role: roleOutput.Role}, err
+}
+
+type role struct {
+	iam *awsprovider.IAM
+	*iam.Role
+}
+
+func (r *role) addRoleToInstanceProfile(ctx context.Context, clusterName string) error {
+	profile, err := createInstanceProfile(ctx, r.iam, KitNodeInstanceProfileNameFor(clusterName))
```

Comment:
could you take `InstanceProfileName` as param to make it compliant with your function name as it's generic.